features/keychain/module/keychain.js

```diff
@@ -44,30 +44,50 @@ export class Keychain {
     this.secp256k1 = secp256k1.create({ getPrivateHDKey: this.#getPrivateHDKey })
   }
 
-  #assertPrivateKeysUnlocked() {
-    assert(!this.#privateKeysAreLocked, 'private keys are locked')
+  #assertPrivateKeysUnlocked(seedIds) {
+    const locked = this.#checkPrivateKeysLocked(seedIds)
+    assert(!locked, 'private keys are locked')
   }
 
-  arePrivateKeysLocked() {
-    return this.#privateKeysAreLocked
+  #checkPrivateKeysLocked(seedIds) {
+    if (!seedIds?.size) {
+      return Object.values(this.#seedLockStatus).some(Boolean)
+    }
+
+    const existingSeeds = Object.entries(this.#seedLockStatus)
+    const checkingSeeds = existingSeeds.filter(([seedId]) => seedIds.has(seedId))
+
+    assert(checkingSeeds.length === seedIds.size, 'must pass in existing seeds')
+    return checkingSeeds.some(([, locked]) => locked)
+  }
+
+  arePrivateKeysLocked(seeds = []) {
```

Comment:
the change is non-breaking for previous consumers because it defaults back to checking that private keys of all seeds are locked, right?

Reply:
Yes, if no seeds are passed, an empty `seedIds` will be passed which gets checked

https://github.com/ExodusMovement/exodus-oss/blob/b93d9802c8146534561dee5b1b75aee370cd7f92/features/keychain/module/keychain.js#L53-L55

Reply:
Can we assert that `Array.isArray(seeds)`?